test/sv/interfaces.mlir

```diff
@@ -0,0 +1,40 @@
+// RUN: circt-opt %s | FileCheck %s
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+// CHECK-LABEL: func @test() {
+func @test() {
+  // Basic interface smoke test
+
+  sv.interface @myinterface {
+    sv.interface.signal "data" : i32
+    sv.interface.modport "input_port" [{direction = "input", signal = @data}]
+    sv.interface.modport "output_port" [{direction = "output", signal = @data}]
+  }
+
+  // CHECK-NEXT: sv.interface @myinterface {
+  // CHECK-NEXT:   sv.interface.signal "data" : i32
+  // CHECK-NEXT:   sv.interface.modport "input_port" [{direction = "input", signal = @data}]
+  // CHECK-NEXT:   sv.interface.modport "output_port" [{direction = "output", signal = @data}]
+  // CHECK-NEXT: }
+
+  // Hanshake-like interface smoke test
+
+  sv.interface @handshake_example {
+    sv.interface.signal "data" : i32
+    sv.interface.signal "valid" : i1
+    sv.interface.signal "ready" : i1
+    sv.interface.modport "dataflow_in" [{direction = "input", signal = @data}, {direction = "input", signal = @valid}, {direction = "output", signal = @ready}]
```

Comment:
I'm not sure how much this matters, but it would be nice to spiff up the syntax to something like:

```
sv.interface.modport "dataflow_in", input @data, input @valid, output @ready
```

This seems pretty straight-forward by parsing while there is a comma.

Reply:
I took a crack at that in the initial implementation here: https://github.com/llvm/circt/pull/73/commits/a4cd78c292b31506e0d764d57ca1e8132e7f925c#diff-4ffc257e0a9bec9e685d0e22331cb80eR159. After some discussion, I switched to use the declarative assembly format, which is what we have in this revision. I think by using the new custom directives, we can keep using the declarative assembly format for the most part, and use a little custom directive to spiff up the syntax here.